src/gui/tray/NotificationCache.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <set>
+#include <QObject>
+
+namespace OCC {
+
+struct Notification
+{
+    QString title;
+    QString message;
+};
+
+class NotificationCache : public QObject
+{
+    Q_OBJECT
+
+public:
+    explicit NotificationCache(QObject *parent = nullptr);
+
+    bool contains(const Notification &notification) const;
+
+    void insert(const Notification &notification);
+
+    void clear();
+
+private:
+    std::set<uint> _notifications;
```

Comment:
@FlexW Just curious, why not use `QSet<uint>`?

Reply:
Because we had the discussion and I thought that we use STL whenever possible :)

Reply:
By moving away from QObject you might want to move to QSet though. Because right now it doesn't make much difference to use one or the other (QObject not being copyable), but without QObject then it's nice to have the implicit sharing and not have to enter into dealing with efficient copies.

I don't think "STL whenever possible" is a good guideline to be honest. Same thing for "Qt containers whenever possible" actually. As can be seen here they are different enough to both have their own pros and cons.